runtime/onert/core/include/exec/Executors.h

```diff
@@ -57,12 +62,22 @@ class Executors
 private:
   void executeEntries(const IODescription &desc);
 
+  ExecutorIndex generateIndex()
+  {
+    // No need to check if there is an entry with _next_index since
+    // _next_index is always ("the highest index in the object map" + 1)
+    if (ExecutorIndex{_next_index}.valid())
+      return ExecutorIndex{_next_index++};
+    else
+      return ExecutorIndex{};
+  }
+
 private:
-  // TODO Use Executor index
-  //      Changing index will effect if/while compile and kernel implementation
-  std::unordered_map<ir::SubgraphIndex, std::unique_ptr<IExecutor>> _executors;
+  std::unordered_map<ExecutorIndex, std::unique_ptr<IExecutor>> _executors;
   // NOTE _model_edges may use different struct type for executor implementation
   std::unique_ptr<ir::ModelEdges> _model_edges;
+  std::unordered_map<ExecutorIndex, std::pair<ir::ModelIndex, ir::SubgraphIndex>> _index_map;
```

Comment:
@hseok-oh What about inverting key and value?
```cc
std::unordered_map<std::pair<ir::ModelIndex, ir::SubgraphIndex>, ExecutorIndex> _index_map;
```

Reply:
If we use index pair as key, we need to introduce hash function for index pair like index's hash function. I don't know it's good solution.

https://github.com/Samsung/ONE/blob/888235ed09309347e1d4455419f65f23545eb4a3/runtime/onert/core/include/util/Index.h#L151-L157

Reply:
@hseok-oh Yes, defining a hash function for two basically integers is not complicated. Do you have any concern what I am missing? I think it is basically 64 bit integer to 32 bit integer hash map.

Reply:
No. You mean hash function like below?

```cpp
template <> struct hash<std::pair<::onert::ir::ModelIndex, ::onert::ir::SubgraphIndex>>
{
  size_t
  operator()(const std::pair<::onert::ir::ModelIndex, ::onert::ir::SubgraphIndex> &pair) const
    noexcept
  {
    return hash<uint64_t>()((uint64_t(pair.first.value()) << 32) | uint64_t(pair.second.value()));
  }
};
```

Then, we may use this pair on `_executors` map directly, and remove `_index_map`, then don't need to introduce `ExecutorIndex`. I'll try this.

Reply:
@hseok-oh Yes, I guess we are on same page. I left more comment in https://github.com/Samsung/ONE/pull/9740#pullrequestreview-1114734308.